reorganized the first section, in line with MEG and EEG spec

robertoostenveld opened the pull request:
I started making the format more consistent with the MEG and EEG sections of the specification. I especially think that the conversion of the EEG spec by @sappelhoff is a good template for the format. Also,many shorter sections of text can be shared (i.e. copied) from the EEG spec, since they should be the same.

robertoostenveld committed:
reorganized the first section, in line with MEG and EEG spec

robertoostenveld commented:
Please consider this instead of PR #1

robertoostenveld committed:
cleaned up come of teh tables (format only, no content changes)

choldgraf commented:
Thanks very much for the PR @robertoostenveld :-) I agree w/ you that it's better to standardize these sections and I agree that @sappelhoff did a great job in that PR, I just hadn't gotten around to taking another pass through this one. I'll give it a closer look today in case I have comments!

robertoostenveld commented:
I could help getting the other sections in the same format as well.

choldgraf commented:
if you have a moment, then please feel free to do so. The PR is currently just my first pass at converting the google doc into markdown format, but I haven't spent a lot of time thinking about overall organization and how it should fit in with the current spec. My time for BIDS this week is gonna be spent on finalizing the draft for submission, so if you have the cycles please feel free to improve this!

robertoostenveld committed:
small formatting, provide link to sourcedata documentation

robertoostenveld committed:
removed physio files from file/directory listing
they is not specifically relevant for iEEG and the iEEG extension does not change them in any way

robertoostenveld committed:
removed surface data derivatives that were mentioned in the file/directory listing
this is not part of the official specification and also not discussed further in the iEEG spec

robertoostenveld committed:
once more removed surface derivatives data, removed CT data (those are explained elsewhere in the BIDS specification), removed comparison to MEG specification

robertoostenveld committed:
start with a short template with only ieeg file and ieeg.json, just like EEG structure.
I have added a header that indicates from where on it needs to be further cleaned up. My plan is to start pushing that header further down, keeping the not-yet-reformatted text below it.

robertoostenveld committed:
split the tables differently, one section is shared with all modalities, the other is iEEG specific (although there is some overlap with EEG and MEG)

robertoostenveld committed:
don't use square brackets in tables to indicate optional items, use (only) the word "OPTIONAL"

robertoostenveld committed:
insert optional ses-label in the file/directory strucure

robertoostenveld committed:
reorganized the must/should/may sections of the gheneric and ieeg specific fields

robertoostenveld committed:
added channel.tsv section, removed space from iEEG data filename (since iEEG data itself is not represented in a specific space), that should be used for positional data (i.e. electrodes)

robertoostenveld committed:
cleaned up channel.tsv section, show example as a text block, not as a table

robertoostenveld committed:
further cleanups to channel section

robertoostenveld committed:
cleaned up the electrodes section

robertoostenveld committed:
cleaned up whilespace in tables

robertoostenveld committed:
cleaned up section on coordsystem.json

robertoostenveld committed:
reorganized section on photo's

robertoostenveld committed:
include template filename for electrodes, coordsystem and photos

robertoostenveld committed:
included section on electrical stimulation

robertoostenveld committed:
wrap the text of some long lines to a decent page width

robertoostenveld commented:
@choldgraf can you please review the result of my reformatting and - if ok - merge it into your `bep010` branch? It will then be part of https://github.com/bids-standard/bids-specification/pull/118 and can be reviewed again by @chrisfilo and others.

robertoostenveld committed:
used https://prettier.io to clean up and reflow the text and make the tables more consistent

choldgraf commented:
Sounds good - I was gonna give it a review but just noticed that you made a commit since your last message...would you like me to hold off a bit? Or ready for review now?

robertoostenveld commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+
+[TODO: brain picture 2]
+
+## Electrical stimulation
```
I think this addresses Q1 of https://github.com/bids-standard/bids-specification/issues/61

robertoostenveld replied:
@dorahermes and @choldgraf are you ok with this, or is more needed on electrical stimulation?

robertoostenveld commented:
sorry, the commit ~20 minutes ago was not planned, just me trying out this prettier.io application that ChrisG mentioned in another PR I was working on. Please go ahead, I will not touch it any more!

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+    by Brain Products GmbH.
 
-Examples of electrical stimulation are further discussed below.
+**Allowed Formats**
```
```suggestion
**Accepted Formats**
```

choldgraf replied:
just for consistency's sake since we use this word above

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+added advantage over the existing formats and should have wide adoption in the
+BIDS-iEEG community.
+
+The original data format is especially valuable in case conversion elicits the
```
```suggestion
The data format in which the data was originally stored is especially valuable in case conversion elicits the
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+loss of crucial metadata specific to manufacturers and specific iEEG systems. We
+also encourage users to provide additional meta information extracted from the
+manufacturer specific data files in the sidecar JSON file. Other relevant files
+MAY be included alongside the original EEG data in the
```
I guess it's fine to say EEG since the intracranial is implied?

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+The original data format is especially valuable in case conversion elicits the
+loss of crucial metadata specific to manufacturers and specific iEEG systems. We
+also encourage users to provide additional meta information extracted from the
+manufacturer specific data files in the sidecar JSON file. Other relevant files
```
```suggestion
manufacturer-specific data files in the sidecar JSON file. Other relevant files
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+MAY be included alongside the original EEG data in the
+[`/sourcedata` directory](../02-common-principles.md#source-vs-raw-vs-derived-data).
+
+For proper documentation of iEEG recording metadata it is important to
```
```suggestion

### Terminology: Electrodes vs. Channels

For proper documentation of iEEG recording metadata it is important to
```
What do you think about this, just to make the page easier to parse?

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+For proper documentation of iEEG recording metadata it is important to
+understand the difference between electrode and channel: An EEG electrode is
+attached to the skin, whereas a channel is the combination of the analog
+differential amplifier and analog-to-digital converter that result in a
```
```suggestion
differential amplifier and analog-to-digital converter that results in a
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+potential (voltage) difference that is stored in the EEG dataset. We employ the
+following short definitions:
+
+-   Electrode = A single point of contact between the acquisition system and the
```
```suggestion
-   **Electrode** = A single point of contact between the acquisition system and the
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+    organized as arrays, grids, leads, strips, probes, shafts, caps (for EEG),
+    etc.
+
+-   Channel = A single analog-to-digital converter in the recording system that
```
```suggestion
-   **Channel** = A single analog-to-digital converter in the recording system that
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+Therefore they are not represented as channels in the data. The type of
+referencing for all channels and optionally the location of the reference
+electrode and the location of the ground electrode MAY be specified.
+### Sidecar JSON document (`*_ieeg.json`)
```
```suggestion

### Sidecar JSON document (`*_ieeg.json`)
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| TaskName           | REQUIRED. Name of the task (for resting state use the “rest” prefix). No two tasks should have the same name. Task label is derived from this field by removing all non alphanumeric ([a-zA-Z0-9]) characters. Note this does not have to be a “behavioral task” that subjects perform, but can reflect some information about the conditions present when the data was acquired (e.g.,  ...
+| SamplingFrequency  | REQUIRED. Sampling frequency (in Hz) of all the iEEG channels in the recording (e.g., 2400). All other channels should have frequency specified as well in the `channels.tsv` file.                                                                                                                                                                                                      ...
+| PowerLineFrequency | REQUIRED. Frequency (in Hz) of the power grid where the iEEG recording was done (i.e., 50 or 60).                                                                                                                                                                                                                                                                                        ...
+| SoftwareFilters    | REQUIRED. List of temporal software filters applied or ideally key:value pairs of pre-applied filters and their parameter values. (n/a if none). E.g., “{'HighPass': {'HalfAmplitudeCutOffHz': 1, 'RollOff: '6dB/Octave'}}”.                                                                                                                                                             ...
```
```suggestion
| SoftwareFilters    | REQUIRED. List of temporal software filters applied or ideally key:value pairs of pre-applied filters and their parameter values. (n/a if none). E.g., `{'HighPass': {'HalfAmplitudeCutOffHz': 1, 'RollOff: '6dB/Octave'}}`.                                                                                                                                                                                            |
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+the values of metadata fields from the actual raw data. Whenever possible,
+please avoid using ad hoc wording.
+
+Generic fields MUST be present:
```
```suggestion
**Generic fields**

MUST be present:
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| PowerLineFrequency | REQUIRED. Frequency (in Hz) of the power grid where the iEEG recording was done (i.e., 50 or 60).                                                                                                                                                                                                                                                                                        ...
+| SoftwareFilters    | REQUIRED. List of temporal software filters applied or ideally key:value pairs of pre-applied filters and their parameter values. (n/a if none). E.g., “{'HighPass': {'HalfAmplitudeCutOffHz': 1, 'RollOff: '6dB/Octave'}}”.                                                                                                                                                             ...
+
+SHOULD be present: For consistency between studies and institutions, we
```
```suggestion
**SHOULD be present**

For consistency between studies and institutions, we
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| InstitutionAddress     | RECOMMENDED. The address of the institution in charge of the equipment that produced the composite instances.                                                                                                                 |
+| DeviceSerialNumber     | RECOMMENDED. The serial number of the equipment that produced the composite instances. A pseudonym can also be used to prevent the equipment from being identifiable, as long as each pseudonym is unique within the dataset. |
+
+Specific iEEG fields MUST be present:
```
```suggestion
**Specific iEEG fields**

MUST be present:
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| :------------ | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| iEEGReference | REQUIRED. General description of the reference scheme used and (when applicable) of location of the reference electrode in the raw recordings (e.g., "left mastoid”, “bipolar”, “T01” for electrode with name T01, “intracranial electrode on top of a grid, not included with data”, “upside down electrode”). If different channels have a different reference, this field should have a ge ...
+
+Specific iEEG fields SHOULD be present:
```
```suggestion
**SHOULD be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| iEEGPlacementScheme             | RECOMMENDED. Freeform description of the placement of the iEEG electrodes. Left/right/bilateral/depth/surface (e.g., “left frontal grid and bilateral hippocampal depth” or “surface strip and STN depth” or “clinical indication bitemporal, bilateral temporal strips and left grid”). |
+| iEEGElectrodeGroups             | RECOMMENDED. Field to describe the way electrodes are grouped into strips, grids or depth probes e.g., {'grid1': "10x8 grid on left temporal pole", 'strip2': "1x8 electrode strip on xxx"}.                                                                                             |
+
+Specific iEEG fields MAY be present:
```
```suggestion
**MAY be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
-1. All specifications of paths need to use forward slashes.
-2. The inheritance principle applies: any metadata file (.json, `.tsv`, etc.) may be defined at any directory level. The values from the top level are inherited by all lower levels unless they are overridden by a file at the lower level. For details see fMRI-BIDS section 3.5:
-(TODO: link to spec)
+The columns of the Channels description table stored in \*\_channels.tsv are:
```
```suggestion
The columns of the Channels description table stored in `*_channels.tsv` are listed below.
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+The columns of the Channels description table stored in \*\_channels.tsv are:
 
-### File formats for the raw data
+MUST be present:
```
```suggestion
**MUST be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| high_cutoff | REQUIRED. Frequencies used for the high pass filter applied to the channel in Hz. If no high pass filter applied, use `n/a`.                                                                                                                                                                                                                                 |
 
-The allowed file formats are divided into two groups: *recommended* formats that are open, well-defined standards, and *accepted* formats that are common in the community. In general, it is discouraged to use *accepted* formats over *recommended* formats, particularly because there are conversion scripts available in most analytics languages to convert data into *recommended* formats. Below are l ...
+SHOULD be present:
```
```suggestion
**SHOULD be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
````diff
@@ -2,76 +2,501 @@
-  Included because it can include FDA-approved encryption of patient information in video format.
+```Text
+name  type  units sampling_frequency low_cutoff high_cutoff notch group reference description              status status_description
+LT01  ECOG  μV    10000             3000       0.11        n/a   LTG   mastoid   lateral_temporal_surface good   n/a
````
```suggestion
LT01  ECOG  μV    10000              3000       0.11        n/a   LTG   mastoid   lateral_temporal_surface good   n/a
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
````diff
@@ -2,76 +2,501 @@
+```Text
+name  type  units sampling_frequency low_cutoff high_cutoff notch group reference description              status status_description
+LT01  ECOG  μV    10000             3000       0.11        n/a   LTG   mastoid   lateral_temporal_surface good   n/a
+LT02  ECOG  μV    10000              3000       0.11        n/a   LTG   mastod   lateral_temporal_surface bad    brokenn
````
```suggestion
LT02  ECOG  μV    10000              3000       0.11        n/a   LTG   mastod    lateral_temporal_surface bad    broken
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
````diff
@@ -2,76 +2,501 @@
+TR1   TRIG  n/a   1000               n/a        n/a         n/a   5     n/a       ana_trigger              good   n/a
+```
+Restricted keyword list for field type in alphabetic order (shared with the MEG
+and EEG modality; however, only iEEG specific types are listed here):
````
It seems like we're listing more than just the iEEG specific types, no? Perhaps we should put them in bold? Or remove the non-iEEG specific types (like `EEG`)?

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+File that gives the location, size and other properties of iEEG electrodes. Note
+that coordinates are expected in cartesian coordinates according to the
+`iEEGCoordinateSystem` and `iEEGCoordinateSystemUnits` fields in
+`*_coordsystem.json`. **If an `*_electrodes.tsv` file is specified, a
```
```suggestion
`*_coordsystem.json`.

**If an `*_electrodes.tsv` file is specified, a
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+that coordinates are expected in cartesian coordinates according to the
+`iEEGCoordinateSystem` and `iEEGCoordinateSystemUnits` fields in
+`*_coordsystem.json`. **If an `*_electrodes.tsv` file is specified, a
+`*_coordsystem.json` file MUST be specified as well**. The order of the required
```
```suggestion
`*_coordsystem.json` file MUST be specified as well**.

The order of the required
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+`*_coordsystem.json` file MUST be specified as well**. The order of the required
+columns in the `*_electrodes.tsv` file MUST be as listed below.
+
+MUST be present:
```
```suggestion
**MUST be present**

```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| z          | REQUIRED. Z position. If electrodes are in 2D space this should be a column of n/a values.                                                                                   |
+| size       | REQUIRED. Surface area of the electrode, in mm^2.                                                                                                                            |
+
+SHOULD be present:
```
```suggestion
**SHOULD be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+| group        | OPTIONAL. Optional field to specify the group that the electrode is a part of. Note that any group specified here should match a group specified in `_channels.tsv`. |
+| hemisphere   | OPTIONAL. Optional field to specify the hemisphere in which the electrode is placed, one of [‘L’ or ‘R’] (use capital).                                              |
+
+MAY be present:
```
```suggestion
**MAY be present**
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+
+It is preferred that electrodes are localized in a 3D coordinate system (with
+respect to a pre- and/or post-operative anatomical MRI or CT scans or in a
+standard space as specified in BIDS Appendix VIII: preferred names of Coordinate
```
```suggestion
standard space as specified in [BIDS Appendix VIII: preferred names of Coordinate
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+It is preferred that electrodes are localized in a 3D coordinate system (with
+respect to a pre- and/or post-operative anatomical MRI or CT scans or in a
+standard space as specified in BIDS Appendix VIII: preferred names of Coordinate
+systems, such as ACPC).
```
```suggestion
systems](../99-appendices/08-coordinate-systems), such as ACPC).
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+### Allowed 2D coordinate systems
+
+If electrodes are localized in 2D space (only x and y are specified and z is
+n/a), then the positions in this file must correspond to the locations expressed
```
```suggestion
`n/a`), then the positions in this file must correspond to the locations expressed
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+in pixels on the photo/drawing/rendering of the electrodes on the brain. In this
+case, coordinates must be (row,column) pairs, with (0,0) corresponding to the
+upper left pixel and (N,0) corresponding to the lower left pixel.
+### Multiple coordinate systems
```
```suggestion

### Multiple coordinate systems
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+and `[_proc-<label>]` fields.
+
+The optional space label (`[_space-<label>]_electrodes.tsv`) indicates the way
+in which electrode positions are interpreted, see [TODO: link] BEP003 - Common
```
```suggestion
in which electrode positions are interpreted, see [BEP003 - Common
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+
+The optional space label (`[_space-<label>]_electrodes.tsv`) indicates the way
+in which electrode positions are interpreted, see [TODO: link] BEP003 - Common
+Derivatives. Examples include:
```
```suggestion
Derivatives](../06-extensions). Examples include:
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
````diff
@@ -2,76 +2,501 @@
+```
+
+These can include photos of the electrodes on the brain surface, photos of
+anatomical features or landmarks (such as sulcal structure), fiducials. Photos
````
```suggestion
anatomical features or landmarks (such as sulcal structure), and fiducials. Photos
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+
+The session label may be used to specify when the photo was taken.
+
+Example of the operative photo of ECoG electrodes (here an annotated example in
```
```suggestion
Example of the operative photo of ECoG electrodes (here is an annotated example in
```

choldgraf commented on src/04-modality-specific-files/06-intracranial-electroencephalography.md:
```diff
@@ -2,76 +2,501 @@
+onset, duration), the `_events.tsv` file can contain the electrical stimulation
+parameters in addition to other events. Note that these can be intermixed with
+other task events. Electrical stimulation parameters can be described in columns
+called electrical*stimulation*<label>, with labels chosen by the researcher and
```
```suggestion
called `electrical*stimulation*<label>`, with labels chosen by the researcher and
```

choldgraf reviewed (changes requested):
Took a pass through these changes - in general I think that the text and layout are both much improved, thanks very much! I added a few comments and suggested changes here and there. Let me know what you think about those and we can merge once these are resolved.

I didn't do anything with the text underneath your `=====` message at the bottom...I think that'll be easier to look through outside of the GitHub UI so we can spot-check that in the other PR.

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

choldgraf committed:
Update src/04-modality-specific-files/06-intracranial-electroencephalography.md

Co-Authored-By: robertoostenveld <r.oostenveld@gmail.com>

robertoostenveld committed:
use iEEG instead of EEG in text

robertoostenveld commented:
I think I resolved all comments. I also deleted the remainders section and pasted it into a separate document. See https://docs.google.com/document/d/1Ab-cmX5FkxechrOR8ZgnBZsRrxDwcHMKunWLx_KQPZQ/edit?usp=sharing

I propose you merge my PR with your branch, so that others can see this in your https://github.com/bids-standard/bids-specification/pull/118. 

@sappelhoff, is it an idea that you merge the BEP010 branch PR for iEEG with BEP006 PR for EEG, so that a single branch remains to be merged with the main spec? I think it will be easier to finalize the two modality-specific sections (and some updates to MEG) if all are in a single branch.

robertoostenveld committed:
removed the "remainder" section, and copied it to https://docs.google.com/document/d/1Ab-cmX5FkxechrOR8ZgnBZsRrxDwcHMKunWLx_KQPZQ/edit for the time being

choldgraf commented:
perfect, thanks - I'll merge this in with the branch and we can start iterating in the other PR, this is much better!

@sappelhoff I'm happy to make this PR to your branch if that's a better way to move forward.
